assets/javascripts/discourse/initializers/hook-encrypt-upload.js.es6

```diff
@@ -36,45 +41,9 @@ export default {
           return true;
         }
 
-        const dataPromise = isAnImage(file.name)
-          ? new Promise((resolve, reject) => {
-              const img = new Image();
-              img.onload = () => resolve(img);
-              img.onerror = err => reject(err);
-              img.src = window.URL.createObjectURL(file);
-              uploadsUrl[file.name] = img.src;
-            }).then(img => {
-              const ratio = Math.min(
-                Discourse.SiteSettings.max_image_width / img.width,
-                Discourse.SiteSettings.max_image_height / img.height
-              );
-
-              return {
-                original_filename: file.name,
-                width: img.width,
-                height: img.height,
-                thumbnail_width: Math.floor(img.width * ratio),
-                thumbnail_height: Math.floor(img.height * ratio)
-              };
-            })
-          : Promise.resolve({ original_filename: file.name });
-
-        const decryptedPromise = new Promise((resolve, reject) => {
-          const reader = new FileReader();
-          reader.onloadend = () => resolve(reader.result);
-          reader.onerror = err => reject(err);
-          reader.readAsArrayBuffer(file);
-        });
-
-        const keyPromise = new Promise((resolve, reject) => {
-          window.crypto.subtle
-            .generateKey({ name: "AES-GCM", length: 256 }, true, [
-              "encrypt",
-              "decrypt"
-            ])
-            .then(resolve, reject);
-        });
-
+        const dataPromise = fetchDataPromise(file, uploadsUrl);
```

Comment:
Super Nitpick: This is something I just noticed, but has been this way since I first introduced encrypted uploads. I believe better names would be `metadataPromise` and `fetchMetadata`. I consider "data" to refer to the file contents and "metadata" to be the data (name, width, height, etc.) about the data.

*please ignore this*